### ISSUE TITLE
Adding multi-caller support to SRT listener

### ISSUE THREAD

**Issue by amiller-isp** [Haivision/srt#785]
We have a requirement for an "SRT server" which would listen for connections and then send data to multiple connected callers. From my understanding of the SRT protocol there is no support for a well known port number to handle incoming connections and then open dedicated connections on separate ports for each caller-established session. 

I'm thinking of something like FTP https://en.wikipedia.org/wiki/File_Transfer_Protocol

I've been reviewing the live transmit sample and the protocol spec but it seems like this is not supported currently.

Is there any plans to add this functionality? Could you recommend an approach here? It seems like we could build this on top of the existing SRT APIs but it would be an entirely custom application.

**Comment by MarcCymontkowski** [comment]
Hi Ade, there wasn't until the SRT version 1.3.3 release :-)

I sent this info to some cloud ingest portals:

SRT version 1.3.3 with mbedTLS support has been released today:
https://github.com/Haivision/srt/releases/tag/v1.3.3

Another important new features the new listener callback functionality I mentioned before. This allows the application to react on incoming caller connections and either deny or decline a connection. It also allows to select custom passphrase for different connections.

Example implementation:
https://github.com/Haivision/srt/blob/master/testing/srt-test-live.cpp

What callers need to do to allow this is to encode a user name and a resource name (stream name) into the SRT “Stream ID” field, which is part of the handshake. We created a specification for it, which is very simple and doesn’t mean any protocol changes:
https://github.com/Haivision/srt/blob/master/docs/API.md#stream-id-protocol
